Android/ChorusRFLaptimer/app/src/main/java/app/andrey_voroshkov/chorus_laptimer/MainActivity.java

```diff
@@ -366,6 +375,22 @@ private void ensurePermissions() {
         }
     }
 
+    private  void initBroadcastReceiverForBluetooth() {
+        mBluetoothReceiver = new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                if (BluetoothAdapter.ACTION_STATE_CHANGED.equals(intent.getAction())) {
+                    if(intent.getIntExtra(BluetoothAdapter.EXTRA_STATE, -1) == BluetoothAdapter.STATE_ON) {
+                        bt.runService();
+                    } else {
+                        bt.stopService();
```

Comment:
stopping the service on each state change except ON seems not quite safe to me

Reply:
`BluetoothAdapter.EXTRA_STATE` can be either be `STATE_OFF`, `STATE_TURNING_ON`, `STATE_ON` or `STATE_TURNING_OFF`. In each case except `STATE_ON` we want the service to be stopped to avoid any invalid state and cover all edge cases. It is correct, that we might be stopping the service several times, but stopping an already stopped service shouldn't change the program state at all.

Stopping the service multiple times, seems even more safe to me, so I am not sure what you mean by "seems not quite safe to me"?